src/category_theory/monoidal/category.lean

```diff
@@ -0,0 +1,405 @@
+end
+
+lemma right_unitor_product_aux_triangle (X Y : C) :
+    ((𝟙 X) ⊗ (associator Y (tensor_unit C) (tensor_unit C)).hom) ≫
+    ((𝟙 X) ⊗ (𝟙 Y) ⊗ (left_unitor (tensor_unit C)).hom)
+  = (𝟙 X) ⊗ (right_unitor Y).hom ⊗ (𝟙 (tensor_unit C)) :=
+begin
+  conv_lhs { rw [←interchange_right_identity] },
+  conv_rhs { congr, skip, rw [←monoidal_category.triangle] }
+end
+
+lemma right_unitor_product_aux_square (X Y : C) :
+    (associator X (Y ⊗ (tensor_unit C)) (tensor_unit C)).hom ≫
+    ((𝟙 X) ⊗ (right_unitor Y).hom ⊗ (𝟙 (tensor_unit C)))
+  = (((𝟙 X) ⊗ (right_unitor Y).hom) ⊗ (𝟙 (tensor_unit C))) ≫
+    (associator X Y (tensor_unit C)).hom :=
+begin
+  conv_rhs { rw [associator_naturality] }
+end
+
+lemma right_unitor_product_aux (X Y : C) :
+    ((associator X Y (tensor_unit C)).hom ⊗ (𝟙 (tensor_unit C))) ≫
+    (((𝟙 X) ⊗ (right_unitor Y).hom) ⊗ (𝟙 (tensor_unit C)))
+  = ((right_unitor (X ⊗ Y)).hom ⊗ (𝟙 (tensor_unit C))) :=
+begin
+  rw ←(cancel_mono (associator X Y (tensor_unit C)).hom),
+  slice_lhs 2 3 { rw ←right_unitor_product_aux_square },
+  conv_lhs { congr, skip, congr, skip, rw [←right_unitor_product_aux_triangle] },
+  conv_rhs { rw [←right_unitor_product_aux_perimeter] }
+end
+
+-- See Proposition 2.2.4 of http://www-math.mit.edu/~etingof/egnobookfinal.pdf
+@[simp] lemma left_unitor_product (X Y : C) :
+  ((associator (tensor_unit C) X Y).hom) ≫ ((left_unitor (X ⊗ Y)).hom) =
+    ((left_unitor X).hom ⊗ (𝟙 Y)) :=
+by rw [←tensor_left_iff, interchange_right_identity, left_unitor_product_aux]
+
+@[simp] lemma left_unitor_product_inv (X Y : C) :
+  ((left_unitor (X ⊗ Y)).inv) ≫ ((associator (tensor_unit C) X Y).inv) =
+    ((left_unitor X).inv ⊗ (𝟙 Y)) :=
+eq_of_inv_eq_inv (by simp)
+
+@[simp] lemma right_unitor_product (X Y : C) :
+  ((associator X Y (tensor_unit C)).hom) ≫ ((𝟙 X) ⊗ (right_unitor Y).hom) =
+    ((right_unitor (X ⊗ Y)).hom) :=
+by rw [←tensor_right_iff, interchange_left_identity, right_unitor_product_aux]
+
+@[simp] lemma right_unitor_product_inv (X Y : C) :
+  ((𝟙 X) ⊗ (right_unitor Y).inv) ≫ ((associator X Y (tensor_unit C)).inv) =
+    ((right_unitor (X ⊗ Y)).inv) :=
+eq_of_inv_eq_inv (by simp)
+
+lemma associator_inv_naturality {X Y Z X' Y' Z' : C} (f : X ⟶ X') (g : Y ⟶ Y') (h : Z ⟶ Z') :
+  (f ⊗ (g ⊗ h)) ≫ (associator X' Y' Z').inv = (associator X Y Z).inv ≫ ((f ⊗ g) ⊗ h) :=
+begin
+  apply (cancel_mono (associator X' Y' Z').hom).1,
+  simp only [assoc, comp_id, iso.inv_hom_id],
+  conv_rhs { congr, skip, rw [associator_naturality] },
+  conv_rhs { rw [←category.assoc], congr, rw [iso.inv_hom_id] },
+  conv_rhs { rw [category.id_comp] }
+end
+
+lemma pentagon_inv (W X Y Z : C) :
+  ((𝟙 W) ⊗ (associator X Y Z).inv) ≫ (associator W (X ⊗ Y) Z).inv ≫ ((associator W X Y).inv ⊗ (𝟙 Z))
+    = (associator W X (Y ⊗ Z)).inv ≫ (associator (W ⊗ X) Y Z).inv :=
+begin
+  apply category_theory.eq_of_inv_eq_inv,
+  dsimp,
+  rw [category.assoc, monoidal_category.pentagon]
+end
+
+@[simp] lemma triangle_1 (X Y : C) :
```

Comment:
Can we give these more meaninful names? Maybe `triangle_assoc_comp_left`, or something like that?

Reply:
Sure, sounds good. Renamed.